indigo_social/forms.py

```diff
@@ -44,7 +44,8 @@ def save(self, commit=True):
         super(UserProfileForm, self).save()
         self.instance.user.first_name = self.cleaned_data['first_name']
         self.instance.user.last_name = self.cleaned_data['last_name']
-        self.instance.user.username = self.cleaned_data['username']
+        if 'username' in self.changed_data:
```

Comment:
What if we changed `clean_username` such that it creates a unique username only if the username isn't specified at all? If the given username isn't unique, then it simply shows an error, rather than trying to create one.

In most cases, users will have a username generated automatically when they sign up.